### DISCUSSION TITLE
Why not allow the use of "BrowserNativeObject" in useFieldArray?

### DISCUSSION THREAD

**Discussion by Toshiuk** [react-hook-form/react-hook-form#9831] [General] [answered]
In the `useFieldArray` function, I have noticed that the current implementation only allows for the use of an object and prohibits not only the use of a primitive type but also "BrowserNativeObject" which is Date, FileList and File. However, I believe that the "BrowserNativeObject" should also be allowed as it is an object and can be used effectively in the function.

In light of this, I suggest that the condition for the "useFieldArray" function be changed from "Primitive | BrowserNativeObject" to only "Primitive" in order to allow for more flexibility in its use.

It would be great to hear the thoughts on this suggestion. And if there is any reason, why it is not recommended to use BrowserNativeObject in this case

**Comment by bluebill1049** [comment, marked as answer]
as far as I know, only FleList and File is not allowed due to browser security restriction.
